nodejs/src/apis/RoomService.ts

```diff
@@ -1,103 +1,111 @@
-    if (results.data.length === 0) {
-      const err = new Error(`no room found with passed in name - ${name}`);
-      logger.error("no room found", err);
+    if (!results.data || results.data.length === 0) {
+      const err = new Error(`no Roomfound with passed in name - ${name}`);
+      logger.error("no Roomfound", err);
       throw err;
     }
     return results.data[0];
   }
 
-  async getHlsURL(roomId: string): Promise<string> {
-    const result: HLSRoomState = await this.apiService.get("/meetings/hls/url", {
-      room_id: roomId,
-    });
-    return result.url || "";
+  /**
+   *
+   * @param enabled Enabled status of Room
+   * @returns a `QueryResultsIterator<HMSRoom>` object
+   */
+  getRoomsByEnabledIterable(enabled: boolean): QueryResultsIterator<HMSRoom> {
+    const queryResultsIterable = new QueryResultsIterator<HMSRoom>(
+      (queryParams: Record<string, any>) => this.apiService.get(this.basePath, queryParams),
+      { enabled }
+    );
+    return queryResultsIterable;
   }
 
-  async createRoom(config: CreateRoomConfig): Promise<HMSRoom> {
-    const payload: Record<string, any> = {
-      name: config.name,
-      description: config.description,
-      template_id: config.templateId,
-      region: config.region,
-    };
-    return this.apiService.post(this.basePath, payload);
-  }
+  /**
+   *
+   * @param before Timestamp before which the Rooms were created
+   * @param after Timestamp after which the Rooms were created
+   * @returns a `QueryResultsIterator<HMSRoom>` object
+   */
+  getRoomsByTimeRangeIterable(before?: Date, after?: Date): QueryResultsIterator<HMSRoom> {
+    const timeQueryParams: Record<string, Date> = {};
+    if (before) timeQueryParams["before"] = before;
+    if (after) timeQueryParams["after"] = after;
 
-  async startHLS(config: StartRoomHLSConfig) {
-    const payload: Record<string, any> = {
-      room_id: config.roomId,
-      variants: [{ meeting_url: config.meetingUrl }],
-    };
-    if (config.recording) {
-      payload.recording = {
-        hls_vod: !!config.recording.hlsVod,
-        single_file_per_layer: !!config.recording.singleFilePerLayer,
-      };
-    }
-    if (config.scheduleAt) {
-      payload.schedule_at = Math.floor(config.scheduleAt.getTime() / 1000);
-    }
-    logger.info("starting hls - ", payload);
-    return this.apiService.post("/meetings/hls/start", payload);
+    const queryResultsIterable = new QueryResultsIterator<HMSRoom>(
+      (queryParams: Record<string, any>) => this.apiService.get(this.basePath, queryParams),
+      timeQueryParams
+    );
+    return queryResultsIterable;
   }
 
-  async stopHLS({ roomId }: { roomId: string }) {
-    logger.info("stopping hls for room - ", roomId);
-    await this.apiService.post("/meetings/hls/stop", { room_id: roomId });
+  /**
+   *
+   * @param config Config of the Room to be created
+   * @returns a `HMSRoom` object
+   */
+  async createRoom(config?: HMSCreateRoomConfig): Promise<HMSRoom> {
```

Comment:
btw I don't see much point in prefixing every interface with HMS, I think it just becomes more verbose to read which is why I didn't include the prefix originally. wdyt?

Reply:
I feel like if the user has their own types similar to the ones in the SDK, it would conflict. 
For ex, someone using the SDK has their own room interface `Room` that they use to log in a DB every time a new room is created. Not a great example, but something similar. If our interfaces are prefixed with `HMS-`, it helps differentiate what's from the SDK and what's not. 

Does that sound right?